tests/test_model.py

```diff
@@ -704,6 +705,56 @@ def test_forecasting_model_predict_return_valid_times(self):
             np.testing.assert_array_equal(pred_var.time.values, expected_valid_times)
 
 
+def test_patchwise_prediction():
+    """Test that ``.predict_patch`` runs correctly."""
+
+    patch_size = 0.5
+    stride = 0.15
+
+    da = _gen_data_xr(dict(
+            time=pd.date_range("2020-01-01", "2020-01-31", freq="D"),
+            x1=np.linspace(0, 1, 325),
+            x2=np.linspace(0, 1, 650),
+        ),
+        data_vars=["var"])
+
+    dp = DataProcessor()
+    ds = dp(da)  # Compute normalisation parameters
+
+    tl = TaskLoader(context=da, target=da)
+
+    tasks = tl(
+        "2020-01-01",
+        context_sampling="all",
+        target_sampling="all",
+        patch_strategy="sliding",
+        patch_size=patch_size,
+        stride=stride,
+    )
+
+    model = ConvNP(dp, tl)
+
+    pred = model.predict_patch(
+        tasks=tasks,
+        X_t=da,
+        data_processor=dp,
+    )
+
+    # gridded predictions
+    assert [isinstance(ds, xr.Dataset) for ds in pred.values()]
+    # TODO come back to this, for artificial datasets here, shapes of predictions don't match inputs
```

Comment:
I would prefer if we get to the bottom of this and uncomment the test before submitting.

Reply:
Solved.